scripts/extract_dependencies.py

```diff
@@ -0,0 +1,56 @@
+"""Extract the names int the [project.optional-dependencies] section of a pyproject.toml file.
```

Comment:
Not sure where to put the script.  Let me know whether you prefer it somewhere else.  

I don't like the need for such a script in the 1st place but there seems to be no generic way like `pip install .[all]` when using `pyproject.toml`.  Using this scripts avoids repeating dependencies which would be hard to maintain in the long run, e.g.
```python
   pip install .[mysql,knx,dmx,midi,mqtt,pulse,telegram,file_persistence]
```
So for me the script is the better choice.